src/components/ReceiveTableContainer/ReceiveTable/BoardView/PacketView/MeasurementView/MeasurementView.tsx

```diff
@@ -1,45 +1,46 @@
 import { RootState } from "store";
 import styles from "./MeasurementView.module.scss";
 import { Measurement, isNumericMeasurement } from "common";
-import { useSelector } from "react-redux";
-import { memo } from "react";
+import { useContext, useLayoutEffect, useRef } from "react";
+import { TableContext } from "components/ReceiveTableContainer/ReceiveTable/TableUpdater";
+import { useUpdater } from "./useUpdater";
+
 type Props = {
     measurement: Measurement;
 };
 
-function measurementSelector(state: RootState, id: string): Measurement {
-    return state.measurements[id];
-}
-
 export const MeasurementView = ({ measurement }: Props) => {
-    // const measurement = useSelector((state: RootState) =>
-    //     measurementSelector(state, id)
-    // );
-
     const isNumeric = isNumericMeasurement(measurement);
 
+    const { valueRef } = useUpdater(
+        measurement.id,
+        isNumeric
+            ? measurement.value.average.toFixed(3)
+            : measurement.value.toString()
+    );
+
     return (
-        // <div className={styles.measurementView}>
         <>
             <span className={styles.name}>{measurement.name}</span>
             {isNumeric && (
```

Comment:
u can move this into a new component

Reply:
In this case, I think its it's clearer in this format.